### PR TITLE
Update voltageLevel creation endpoint

### PR DESCRIPTION
**Pull request by ayolab**

### CODE REVIEW

File: src/test/java/org/gridsuite/modification/server/modifications/LineAttachToNewVoltageLevelTest.java [hunk 1]
```diff
@@ -38,8 +40,15 @@ private VoltageLevelCreationInfos getNewVoltageLevel() {
                 .equipmentName("NewVoltageLevel")
                 .nominalVoltage(379.3)
                 .substationId("s1")
-                .busbarSections(List.of(new BusbarSectionCreationInfos("v1bbs", "BBS1", 1, 1)))
-                .busbarConnections(List.of())
+                .nominalVoltage(379.1)
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/modification/server/modifications/LineAttachToNewVoltageLevelTest.java:43]: useless ... nominalVoltage is set just above ...

File: src/test/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevelTest.java [hunk 2]
```diff
@@ -42,12 +44,19 @@ private LineCreationInfos getAttachmentLine(String lineName) {
...
     private VoltageLevelCreationInfos getNewVoltageLevel() {
         return VoltageLevelCreationInfos.builder()
-                .equipmentId("newVlName")
+                .equipmentId("newVoltageLevel")
                 .equipmentName("NewVoltageLevel")
                 .nominalVoltage(379.3)
                 .substationId("s1")
-                .busbarSections(List.of(new BusbarSectionCreationInfos("v1bbs", "BBS1", 1, 1)))
-                .busbarConnections(List.of())
+                .nominalVoltage(379.1)
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevelTest.java:51]: useless ... nominalVoltage is set just above ...

File: src/test/java/org/gridsuite/modification/server/modifications/LineSplitWithNewVoltageLevelTest.java [hunk 3]
```diff
@@ -53,20 +55,27 @@ protected Network createNetwork(UUID networkUuid) {
...
-            .nominalVoltage(379.3)
-            .substationId("s1")
-            .busbarSections(List.of(new BusbarSectionCreationInfos("v1bbs", "BBS1", 1, 1)))
-            .busbarConnections(List.of())
-            .build();
+                .equipmentId("newVoltageLevel")
+                .equipmentName("NewVoltageLevel")
+                .nominalVoltage(379.3)
+                .substationId("s1")
+                .nominalVoltage(379.1)
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/modification/server/modifications/LineSplitWithNewVoltageLevelTest.java:62]: useless ... nominalVoltage is set just above ...

File: src/test/java/org/gridsuite/modification/server/modifications/LineSplitWithVoltageLevelTest.java [hunk 4]
```diff
@@ -51,13 +53,20 @@ protected ModificationInfos buildModification() {
...
-            .nominalVoltage(379.3)
-            .substationId("s1")
-            .busbarSections(List.of(new BusbarSectionCreationInfos("v1bbs", "BBS1", 1, 1)))
-            .busbarConnections(List.of())
-            .build();
+                .equipmentId("newVoltageLevel")
+                .equipmentName("NewVoltageLevel")
+                .nominalVoltage(379.3)
+                .substationId("s1")
+                .nominalVoltage(379.1)
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/modification/server/modifications/LineSplitWithVoltageLevelTest.java:60]: useless ... nominalVoltage is set just above ...

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by FranckLecuyer — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/entities/equipment/creation/VoltageLevelCreationEntity.java [hunk 5]
```diff
@@ -28,37 +31,49 @@
...
+    @Column
+    private int busbarCount;
+
+    @Column
+    private int sectionCount;
+
+    @ElementCollection(fetch = FetchType.EAGER)
     @CollectionTable
-    private List<BusbarSectionCreationEmbeddable> busbarSections;
+    @Enumerated(EnumType.ORDINAL)
```
[the remark this answers is not in the document]
Comment (ayolab) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/creation/VoltageLevelCreationEntity.java]: it doesn't work, Powsybl enums are stored in database as Integers, I removed the annotation as for "EnergySource" in "GeneratorCreation"

File: src/test/java/org/gridsuite/modification/server/modifications/LineAttachToNewVoltageLevelTest.java (hunk 1 above)
Comment (ayolab) [reply src/test/java/org/gridsuite/modification/server/modifications/LineAttachToNewVoltageLevelTest.java]: Done

File: src/test/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevelTest.java (hunk 2 above)
Comment (ayolab) [reply src/test/java/org/gridsuite/modification/server/modifications/LineAttachToVoltageLevelTest.java]: Done

File: src/test/java/org/gridsuite/modification/server/modifications/LineSplitWithNewVoltageLevelTest.java (hunk 3 above)
Comment (ayolab) [reply src/test/java/org/gridsuite/modification/server/modifications/LineSplitWithNewVoltageLevelTest.java]: Done

File: src/test/java/org/gridsuite/modification/server/modifications/LineSplitWithVoltageLevelTest.java (hunk 4 above)
Comment (ayolab) [reply src/test/java/org/gridsuite/modification/server/modifications/LineSplitWithVoltageLevelTest.java]: Done

**Review by YenguiSeddik — APPROVED** [review]

**Pull request merged**